src/python/ensembl/io/genomio/gff3/id_allocator.py

```diff
@@ -0,0 +1,191 @@
+    """Raised when there is a problem with an ID."""
+
+
+@dataclass
+class StableIDAllocator:
+    """Set of tools to check and allocate stable IDs."""
+
+    # Multiple parameters to automate various fixes
+    skip_gene_id_validation: bool = False
+    min_id_length: int = 7
+    current_id_number: int = 0
+    make_missing_stable_ids: bool = True
+    prefix: str = "TMP_"
+    _loaded_ids: Set = field(default_factory=set)
+
+    def generate_gene_id(self) -> str:
+        """Returns a new unique gene stable_id with a prefix.
+
+        The ID is made up of a prefix and a number, which is auto incremented.
+
+        """
+        self.current_id_number += 1
+        new_id = f"{self.prefix}{self.current_id_number}"
+        return new_id
+
+    def is_valid(self, stable_id: str) -> bool:
+        """Checks that the format of a stable ID is valid.
+        Args:
+            stable_id: Stable ID to validate.
+        """
+
+        if self.skip_gene_id_validation:
+            logging.debug(f"Validation deactivated by user: '{stable_id}' not checked")
+            return True
+
+        # Trna (from tRNAscan)
+        if re.search(r"^Trna", stable_id, re.IGNORECASE):
+            logging.debug(f"Stable ID is a tRNA from tRNA-scan: {stable_id}")
+            return False
+
+        # Coordinates
+        if re.search(r"^.+:\d+..\d+", stable_id):
+            logging.debug(f"Stable id is a coordinate: {stable_id}")
+            return False
+
+        # Special characters
+        if re.search(r"[ |]", stable_id):
+            logging.debug(f"Stable id contains special characters: {stable_id}")
+            return False
+
+        # Min length
+        if len(stable_id) < self.min_id_length:
+            logging.debug(f"Stable id is too short (<{self.min_id_length}) {stable_id}")
+            return False
+
+        return True
+
+    @staticmethod
+    def remove_prefix(stable_id: str, prefixes: List[str]) -> str:
+        """Returns the stable ID after removing its prefix (if any).
+        Args:
+            stable_id: Stable ID to process.
+            prefixes: List of prefixes to search for.
+        """
+
+        for prefix in prefixes:
+            if stable_id.startswith(prefix):
+                return stable_id[len(prefix) :]
+        return stable_id
+
+    @staticmethod
+    def generate_transcript_id(gene_id: str, number: int) -> str:
+        """Returns a formatted transcript ID generated from a gene ID and number.
+        Args:
+            gene_id: Gene stable ID.
+            number: Positive number.
+        Raises:
+            ValueError: If the number provided is not greater than zero.
+
+        """
+        if number < 1:
+            raise ValueError("Number has to be a positive integer.")
+
+        transcript_id = f"{gene_id}_t{number}"
+        return transcript_id
+
+    def normalize_cds_id(self, cds_id: str) -> str:
+        """Returns a normaliSed version of the provided CDS ID.
```

Comment:
```suggestion
        """Returns a normalized version of the provided CDS ID.
```
Since you are using the American spelling elsewhere.